compiler/damlc/daml-doc/src/DA/Daml/Doc/Extract/Templates.hs

```diff
@@ -191,14 +195,15 @@ getFields adt =
     [] -> [] -- catching the dummy case here, see above
     _other -> error "getFields: found multiple constructors"
 
-mkChoiceDoc :: MS.Map Typename ADTDoc -> Typename -> ChoiceDoc
-mkChoiceDoc typeMap name =
+mkChoiceDoc :: MS.Map Typename ADTDoc -> MS.Map Typename DDoc.Type -> Typename -> ChoiceDoc
+mkChoiceDoc typeMap choiceTypeMap name =
   ChoiceDoc
     { cd_name = ad_name choiceADT
     , cd_descr = ad_descr choiceADT
   -- assumes exactly one constructor (syntactic in the template syntax), or
   -- uses a dummy value otherwise.
     , cd_fields = getFields choiceADT
+    , cd_type = fromMaybe (TypeApp Nothing (Typename "UnknownType") []) $ MS.lookup name choiceTypeMap
```

Comment:
This shouldn't ever happen, as it means a module has a choice without a HasExercise instances.
We could show the Unit type here (`TypeTuple []`), or just a flat call to `error`